cpp/pagerank_module/algorithm_online/pagerank.hpp

```diff
@@ -0,0 +1,154 @@
+#pragma once
+
+#include <random>
+#include <unordered_set>
+
+namespace pagerank_online_alg {
+namespace {
+class PageRankData {
+  ///
+  ///@brief Context for storing data for dynamic pagerank
+  ///
+  ///
+ public:
+  void Init() {
+    walks.clear();
+    walks_counter.clear();
+    walks_table.clear();
+
+    gen = new std::mt19937(42);
+    distr = new std::uniform_real_distribution<float>(0.0, 1.0);
+  }
+
+  bool IsEmpty() { return walks.empty(); }
+
+  /// Keeping the information about walks on the graph
+  std::vector<std::vector<std::uint64_t>> walks;
+
+  // Keeping the information of walk appearance in algorithm for faster calculation
+  std::unordered_map<std::uint64_t, uint64_t> walks_counter;
+
+  /// Table that keeps the node appearance and walk ID
+  std::unordered_map<std::uint64_t, std::unordered_set<std::uint64_t>> walks_table;
+
+  std::mt19937 *gen;
+  std::uniform_real_distribution<float> *distr;
+};
+
+extern PageRankData context;
+extern std::uint64_t global_R;
+extern double global_epsilon;
```

Comment:
Why are they global and not just static inside the `cpp` file. Who else needs to be aware of these variables?

Reply:
At one point I expected `module` to be able to get the context information, however I can change that.

Reply:
I would remove this.